pkg/controller/populators/populator-base.go

```diff
@@ -268,12 +288,17 @@ func (r *ReconcilerBase) reconcile(req reconcile.Request, populator populatorCon
 	}
 
 	// Each populator reconciles the target PVC in a different way
-	if cc.IsUnbound(pvc) || !cc.IsPVCComplete(pvc) || cc.IsMultiStageImportInProgress(pvc) {
-		return populator.reconcileTargetPVC(pvc, pvcPrime)
+	res, err := populator.reconcileTargetPVC(pvc, pvcPrime)
```

Comment:
I think it has to run on every reconcile cycle, because otherwise the import-populator will not be able to retry setting all annotations and labels on the target PVC if import-controller modified the target PVC at the same time and the API refuses to accept the older object on update.

Reply:
I agree with @alromeros, I would feel much better if we only go in here on specific occasions such as the labels not being there

We have a bigger problem here that can be taken outside of the PR

Reply:
@mhenriks Would you be okay with leaving it as it is right now? The tests do not flake anymore and to me it looks correct. IMO adding a short-circuit workaround will be unnecessarily complex.

Reply:
@alromeros @mhenriks if you're both okay with this, I would be too. Just making sure we don't get hotloops over this, similar to https://github.com/kubevirt/containerized-data-importer/pull/2985

Reply:
I have similar concerns, but other than that everything is looking good to me. Since all these populator changes are required due to `updatePVCWithPVCPrimeAnnotations` failing with the `failed to update object` error we can improve this a bit when changing the `update` to `patch` (as I plan to do in the near future).